gluish/common.py

```diff
@@ -64,3 +130,215 @@ def run(self):
 
     def complete(self):
         return which(self.name) is not None
+
+
+class Available(luigi.Task):
+    """
+    Checks, whether an HTTP service is available or not. This task will consider
+    itself complete, only if the HTTP service return a 200.
+    """
+    service = luigi.Parameter()
+    message = luigi.Parameter(default="")
+
+    def run(self):
+        """ Only run if, task is not complete. """
+        raise RuntimeError('HTTP service %s is not available (%s)' % (self.service, self.message))
+
+    def complete(self):
+        return check(self.service) is not None
+
+
+class GitCloneRepository(luigi.Task):
+    """
+    Checks, whether a certain directory already exists (that should contain a Git repository) - otherwise it will be cloned.
+    """
+    gitrepository = luigi.Parameter()
+    repositorydirectory = luigi.Parameter()
+    basedirectory = luigi.Parameter()
+
+    def requires(self):
+        return Executable(name='git')
+
+    def run(self):
+        currentdir = os.curdir
```

Comment:
Temporary directory change appears more than once. Might be a candidate for a context manager. Something like this:

```python
import os
 
class chdir(object):
    """
    Change directory temporarily.
    """
    def __init__(self, path):
        self.wd = os.getcwd()
        self.dir = path
 
    def __enter__(self):
        os.chdir(self.dir)
 
    def __exit__(self, *args):
        os.chdir(self.wd)
```

Which could be used:

```python
def run(self):
    self.output().makedirs()
    with chdir(str(self.basedir)):
        shellout("...")
```